### PR TITLE
Removing the deprecated Named (Native) Queries in Metadata/EntityRepository

### PR DESCRIPTION
**Pull request by sir-kain**
Following [this issue](https://github.com/doctrine/orm/pull/8593), I'm removing the deprecated Named (Native) Queries in Metadata/EntityRepository

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Functional/Ticket/DDC1404Test.php [hunk 1]
```diff
@@ -40,12 +40,10 @@ protected function setUp(): void
...
     {
-        $this->expectDeprecationWithIdentifier('https://github.com/doctrine/orm/issues/8592');
-
         $repository  = $this->_em->getRepository(DDC1404ChildEntity::class);
-        $queryAll    = $repository->createNamedQuery('all');
-        $queryFirst  = $repository->createNamedQuery('first');
-        $querySecond = $repository->createNamedQuery('second');
+        $queryAll    = $repository->createQueryBuilder('p')->getQuery();
+        $queryFirst  = $repository->createQueryBuilder('p')->where('p.id = 1')->getQuery();
+        $querySecond  = $repository->createQueryBuilder('p')->where('p.id = 2')->getQuery();
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/DDC1404Test.php]: You can remove this file entirely, as it was introduced in 8af0f9d0711a7dcba8364e86c29cc3cc6e426010

The title of that commit says "added support for Inherited Named Queries" so that test is meant to test named query.

**Comment by greg0ire** [comment]
Please add an entry in `UPGRADE.md` to let the world know you have removed named queries. Also, there are docs that need to be removed:

```
rg -i 'named.*quer' docs
docs/en/reference/annotations-reference.rst
67:-  :ref:`@NamedNativeQuery <annref_named_native_query>`
882:.. _annref_named_native_query:
884:@NamedNativeQuery
889:    Named Native Queries are deprecated as of version 2.9 and will be removed in ORM 3.0
891:Is used to specify a native SQL named query.
892:The NamedNativeQuery annotation can be applied to an entity or mapped superclass.
912:     * @NamedNativeQueries({
913:     *      @NamedNativeQuery(
1186:     * @NamedNativeQueries({
1187:     *      @NamedNativeQuery(
1192:     *      @NamedNativeQuery(

docs/en/reference/php-mapping.rst
88:-   ``addNamedQuery($name, $dqlQuery)``

docs/en/reference/native-sql.rst
435:Named Native Query
440:    Named Native Queries are deprecated as of version 2.9 and will be removed in ORM 3.0
442:You can also map a native query using a named native query mapping.
445:using named native query (and sql resultset mappings if is a several resultset mappings).
447:Like named query, a named native query can be defined at class level or in an XML file.
450:A resultSetMapping parameter is defined in @NamedNativeQuery,
460:         * @NamedNativeQueries({
461:         *      @NamedNativeQuery(
521:                <named-native-queries>
522:                    <named-native-query name="fetchMultipleJoinsEntityResults" result-set-mapping="mappingMultipleJoinsEntityResults">
524:                    </named-native-query>
525:                </named-native-queries>
555:the ``fetchJoinedAddress`` named query use the joinMapping result set mapping.
568:             * @NamedNativeQueries({
569:             *      @NamedNativeQuery(
607:                <named-native-queries>
608:                    <named-native-query name="findAll" result-set-mapping="mappingFindAll">
610:                    </named-native-query>
611:                </named-native-queries>
636:             * @NamedNativeQueries({
637:             *      @NamedNativeQuery(
687:                <named-native-queries>
688:                    <named-native-query name="fetchJoinedAddress" result-set-mapping="mappingJoinedAddress">
690:                    </named-native-query>
691:                </named-native-queries>
718:             * @NamedNativeQueries({
719:             *      @NamedNativeQuery(
735:                <named-native-queries>
736:                    <named-native-query name="find-by-id" result-class="Address">
738:                    </named-native-query>
739:                </named-native-queries>
755:             * @NamedNativeQueries({
756:             *      @NamedNativeQuery(
782:                <named-native-query name="count" result-set-mapping="mappingCount">
784:                </named-native-query>
```

File: tests/Doctrine/Tests/Models/ReadonlyProperties/Book.php [hunk 2]
```diff
@@ -5,7 +5,6 @@
 namespace Doctrine\Tests\Models\ReadonlyProperties;
 
 use Doctrine\Common\Collections\ArrayCollection;
-use Doctrine\Common\Collections\Collection;
```
Comment (greg0ire) [inline tests/Doctrine/Tests/Models/ReadonlyProperties/Book.php]: Why did you remove this? I think that's what makes the tests fail.

Comment (sir-kain) [reply tests/Doctrine/Tests/Models/ReadonlyProperties/Book.php]: This line is removed automatically by `phpcbf`

Comment (greg0ire) [reply tests/Doctrine/Tests/Models/ReadonlyProperties/Book.php]: Indeed, I experience the same bug when I try your branch. Can you try restoring that line and pushing? Maybe it will work in the CI, as the CI uses PHP 8.1.

File: psalm-baseline.xml [hunk 3]
```diff
@@ -487,15 +487,6 @@
       <code>matching</code>
     </UndefinedInterfaceMethod>
   </file>
-  <file src="lib/Doctrine/ORM/Mapping/Builder/ClassMetadataBuilder.php">
-    <ArgumentTypeCoercion occurrences="2">
-      <code>$class</code>
-      <code>$repositoryClassName</code>
-    </ArgumentTypeCoercion>
```
Comment (greg0ire) [inline psalm-baseline.xml:494]: You should restore this

File: psalm-baseline.xml [hunk 4]
```diff
@@ -487,15 +487,6 @@
...
-    <DeprecatedMethod occurrences="1">
-      <code>addNamedQuery</code>
-    </DeprecatedMethod>
```
Comment (greg0ire) [inline psalm-baseline.xml:497]: But this should indeed be removed.

File: psalm-baseline.xml (hunk 3 above)
Comment (sir-kain) [reply psalm-baseline.xml:494]: Ok, we still need this rule `ArgumentTypeCoercion` 👍🏾

File: psalm-baseline.xml [hunk 5]
```diff
@@ -1048,14 +1014,6 @@
...
     </PossiblyFalseOperand>
   </file>
-  <file src="lib/Doctrine/ORM/NativeQuery.php">
-    <LessSpecificImplementedReturnType occurrences="1">
-      <code>mixed</code>
-    </LessSpecificImplementedReturnType>
-    <PropertyNotSetInConstructor occurrences="1">
-      <code>$sql</code>
-    </PropertyNotSetInConstructor>
-  </file>
```
Comment (greg0ire) [inline psalm-baseline.xml]: This block should also be restored

File: tests/Doctrine/Tests/Models/ReadonlyProperties/Book.php (hunk 2 above)
Comment (sir-kain) [reply tests/Doctrine/Tests/Models/ReadonlyProperties/Book.php]: OK lets see with the CI

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @sir-kain !
